src/plugins/vis_type_vislib/public/line_to_expression.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Any modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// import { get } from 'lodash';
+import { buildVislibDimensions, Vis } from '../../visualizations/public';
+import { buildExpression, buildExpressionFunction } from '../../expressions/public';
+import { OpenSearchaggsExpressionFunctionDefinition } from '../../data/common/search/expressions';
+import {
+  VegaExpressionFunctionDefinition,
+  VegaSpecExpressionFunctionDefinition,
+} from '../../vis_type_vega/public';
+
+export const toExpressionAst = async (vis: Vis, params: any) => {
+  // Construct the existing expr fns that are ran for vislib line chart, up until the render fn.
+  // That way we get the exact same data table of results as if it was a vislib chart.
+  const opensearchaggsFn = buildExpressionFunction<OpenSearchaggsExpressionFunctionDefinition>(
+    'opensearchaggs',
+    {
+      index: vis.data.indexPattern!.id!,
+      metricsAtAllLevels: vis.isHierarchical(),
+      partialRows: vis.params.showPartialRows || false,
+      aggConfigs: JSON.stringify(vis.data.aggs!.aggs),
+      includeFormatHints: false,
```

Comment:
where is `includeFormatHints` coming from? I don't see it in build_pipeline

Reply:
I set it to false because I saw it was setup like so everywhere this was being used.
reference: https://github.com/search?q=repo%3Aopensearch-project%2FOpenSearch-Dashboards%20%22buildExpressionFunction%3COpenSearchaggsExpressionFunctionDefinition%3E%22&type=code

Reply:
I see in `opensearchaggs` fn definition it is defaulted to false, so sounds good.